src/plugin.py

```diff
@@ -116,56 +122,56 @@ async def pass_login_credentials(self, credentials: Dict[str, str], _ : List[Dic
         elif login_state == WebpageView.PARANOID_ENCIPHERED:
-            return await self._handle_manual_eciphering_result(credentials)
+            return await self._handle_manual_enciphering_result(credentials)
         else:
-            logger.error("Unexpected state in pass_login_credentials")
+            logger.error(f"Unexpected state {login_state:r} in pass_login_credentials")
             raise UnknownBackendResponse()
 
-    async def _handle_login_result(self, credentials: Dict[str, str]) -> Union[Authentication, NextStep]: #todo revert when not testing.
+    async def _handle_login_result(self, credentials: Dict[str, str]) -> Union[Authentication, NextStep]:  # todo revert when not testing.
         self._use_paranoid_login = False
 
         data_or_error = self._view.retrieve_data_regular_login(credentials)
-        print("Got regular login data")
+        logger.debug("Got regular login data")
         if isinstance(data_or_error, NextStep):
             return data_or_error
-        (username, password) = data_or_error
-        print("waiting for rsa key")
-        key_or_error = await self._model.retrieve_rsa_key(username) #revert
+
+        username, password = data_or_error
+
+        logger.debug("waiting for rsa key")
+        key_or_error = await self._model.retrieve_rsa_key(username)  # revert
+
         if isinstance(key_or_error, SteamPublicKey):
             logger.info("received new RSA key from steam")
-            print("received new RSA key from steam")
             key = cast(SteamPublicKey, key_or_error)
             enciphered = encrypt(password.encode('utf-8', errors="ignore"), key.rsa_public_key)
             return await self._do_common_credential_login(username, enciphered, key.timestamp, False)
-        else:
-            logger.warning("Login failed on the rsa key. this is an unexpected behavior.")
-            print("Login failed on the rsa key. this is an unexpected behavior.")
-            return self._view.login_failed(cast(ModelAuthError, key_or_error))
+
+        logger.warning("Login failed on the rsa key. this is an unexpected behavior.")
+        return self._view.login_failed(key_or_error)
 
     async def _handle_two_factor_code_result(self, credentials: Dict[str, str], is_email_code: bool) -> Union[Authentication, NextStep]:
-        
-        if (self._unauthed_steam_id is None or self._two_factor_info is None):
+        if self._unauthed_steam_id is None or self._two_factor_info is None:
             logger.exception("Two Factor page returned but the steam id and two factor information are not initialized. This is unexpected")
             raise UnknownBackendResponse()
 
         code_or_error = self._view.retrieve_data_two_factor(credentials, self._two_factor_info.allowed_authentication_methods, self._use_paranoid_login)
-        
-        if(isinstance(code_or_error, NextStep)):
+
+        if isinstance(code_or_error, NextStep):
             return code_or_error
-        else:
```

Comment:
I just want to make the code more readable by removing several levels of indentation, that's all. I have a strong personal opinion on indentation levels in Python.

Also, let's be real: after what we've experienced in the past few weeks I think it's well advised to assume that the next iteration of the authentication workflow requires another rewrite anyway. So let's do this one properly so that we have a clean start for if we need to rewrite it.

Reply:
Fair play.